plugins/services/src/js/components/modals/ServiceRootGroupModal/index.tsx

```diff
@@ -0,0 +1,479 @@
+          <Trans>Advanced Settings</Trans>
+        </AdvancedSectionLabel>
+        <AdvancedSectionContent>
+          <FormRow>
+            <FormGroup className="column-12">
+              <FieldLabel>
+                <FormGroupHeading>
+                  <FormGroupHeadingContent>
+                    <Trans>Role Enforcement</Trans>
+                  </FormGroupHeadingContent>
+                  <FormGroupHeadingContent>
+                    <Tooltip
+                      content={roleEnforcementTooltipContent}
+                      interactive={true}
+                      maxWidth={300}
+                      wrapText={true}
+                    >
+                      <InfoTooltipIcon />
+                    </Tooltip>
+                  </FormGroupHeadingContent>
+                </FormGroupHeading>
+              </FieldLabel>
+              <FieldLabel className="text-align-left">
+                <FieldInput
+                  checked={data.enforceRole}
+                  type="radio"
+                  name="enforceRole"
+                  value={true}
+                />
+                <Trans>Use Group Role</Trans>
+                <SpacingBox side="left" spacingSize="s" tag="span">
+                  <Badge>
+                    <Trans>Recommended</Trans>
+                  </Badge>
+                </SpacingBox>
+                <FieldHelp>
+                  <Trans>
+                    Allows Quota to be enforced on all the services in the
+                    group.
+                  </Trans>
+                </FieldHelp>
+              </FieldLabel>
+              <FieldLabel className="text-align-left">
+                <FieldInput
+                  checked={!data.enforceRole}
+                  type="radio"
+                  name="enforceRole"
+                  value={false}
+                />
+                <Trans>Use Legacy Role</Trans>
+                <FieldHelp>
+                  <Trans>
+                    Will not enforce quota on all services in the group.
+                  </Trans>
+                </FieldHelp>
+              </FieldLabel>
+            </FormGroup>
+          </FormRow>
+        </AdvancedSectionContent>
+      </AdvancedSection>
+    );
+  }
+
+  handleFormChange(event: React.FormEvent<HTMLFormElement>) {
+    if (this.state.isPending) {
+      return;
+    }
+    const target = event.target as HTMLInputElement;
+    const fieldName = target.getAttribute("name");
+    if (!fieldName) {
+      return;
+    }
+    let value: string | boolean;
+    switch (target.type) {
+      case "checkbox":
+        value = target.checked;
+        break;
+      case "radio":
+        value = target.value;
+        if (value === "true" || value === "false") {
+          value = value === "true";
+        }
+        break;
+      default:
+        value = target.value;
+        break;
+    }
+
+    const { data } = this.state;
+    this.setState({ data: set(data, fieldName, value) });
```

Comment:
For fun I took a stab at writing a version of set that does not require a dependency (lodash.set is 5KB minified)

```typescript
function set(object: any, path: string, value: any): any {
  if (typeof object !== "object") {
    return object;
  }
  const delimiterIndex = path.indexOf(".");
  if (delimiterIndex >= 0) {
    const thisPart = path.substring(0, delimiterIndex);
    object[thisPart] = set(
      object[thisPart] || {},
      path.substring(delimiterIndex + 1),
      value
    );
  } else {
    object[path] = value;
  }
  return object;
}

let obj = set({}, "quota.cpus", 1);
set(obj, "quota.superman", 1);
set(obj, "quota.cpus.whatHappens", 0);

console.log(obj);
// { quota: { cpus: 1, superman: 1 } }
```

Reply:
I initially had a function written for this and opt-ed for `lodash.set` instead.

Reply:
I switched to lodash b/c it seemed like a common enough use case that it had to of been solved before, I found the lodash function and opted for it since it's less code for use to maintain/test etc. But that is a total preference thing.